### DISCUSSION TITLE
3UV DHW defekt?

### DISCUSSION THREAD

**Discussion by DanieleNeo** [Trunks1982/Daikin-Rotex-HPSU-CAN#64] [General]
Hi,

while monitoring the operation of the heat pump I noticed that this error is often displayed even though the pump seems to be working correctly and does not show any errors on its monitor. This is the section of code that display it

![image](https://github.com/user-attachments/assets/38ad5865-211e-4d6e-a615-55ca6b30b521)

And this a piece of log

`[01:14:41][I][unhandled:146]: can_id<0x10A> data<31 00 FA C0 B4 00 00>
[01:14:41][I][unhandled:146]: can_id<0x180> data<22 0A FA C0 B4 00 22>
[01:14:42][I][sendGet:146]: Error Code can_id<0x680> command<31 00 FA 13 88 00 00>
[01:14:42][E][daikin_rotex_can:313]: 3UV DHW defekt => tvbh: 33.799999, tv: 28.400000, max_spread: 3.000000, bpv: 0.000000, flow_rate: 1424.000000
[01:14:42][I][handle :146]: Error Code<No error|3UV DHW defekt> can_id<0x180> data<D2 00 FA 13 88 00 00> changed<0>
[01:14:42][I][sendGet:146]: Operating Mode can_id<0x680> command<31 00 FA 01 12 00 00>
[01:14:42][I][handle :146]: Operating Mode<Heating> can_id<0x180> data<D2 00 FA 01 12 03 00> changed<0>
[01:14:42][I][sendGet:146]: ES mode can_id<0x680> command<31 00 FA 01 76 00 00>
[01:14:42][I][handle :146]: ES mode<> can_id<0x180> data<D2 00 FA 01 76 00 00> changed<0>`

what is this error for?

Thanks in advance,
D.

**Comment by wrfz** [comment]
The air source heat pumps have two 3-way valve switches, which are relatively prone to failure. Usually, they break after 2-4 years. Over time, they calcify and become stiffer, sometimes eventually getting completely stuck. This causes the stepper motor to no longer be able to fully open or close the valve. As a result, the hot water from the hot water tank is quickly diverted to the floor, which is undesirable given the relatively expensive water due to the high flow temperature. The air source heat pump cannot detect this defect. Therefore, we have implemented detection in the ESPHome component, which performs a plausibility check based on the Vorlauf, VorlaufBH, Rücklauf, Durchfluss, and Mixer position or BPV position.

However, in your case, the code is somewhat outdated. There was a bug in the code that incorrectly throws an error, even though the valve may still be functioning correctly.

Please flash the latest version of the ESPHome CAN component! If the error occurs again, please report it again. Then we can check certain values to determine if the valve is actually defective.

**Reply by DanieleNeo** [reply]
Hi @wrfz,

thanks for your comprehensive answer. Probably as you say it is a false error due to the code not updated, the heat pump although from 2015 was put into operation this year and has just 1 month of operational life, so I really think it does not have a problem with the valves, also because as I said the machine does not report errors and does not highlight problems.

Yes, I saw that there have been corrections on the repository, I forked both (can and serial) because I am working on an English/Italian regionalization, as already said I find your work wonderful but unfortunately I have no ability with German. I have to align the forked repository and verify if the problem is actually solved, if not I will definitely let you know.

Two questions a little abstract from the topic, but since you are experts in rotex heat pumps (since you have made a monitoring and control software) I imagine you can give me some feedback: I noticed that when temperatures are cold the machine performs several defrost cycles, I know perfectly well that this is an absolutely normal thing, the problem however is that it does it often, one every hour, each time subtracting about 2 degrees from the technical accumulation and triggering DHW recharges at night when perhaps the situation is not the best for this type of activity (external temperature at zero or less degrees)

![image](https://github.com/user-attachments/assets/1b5f7f7e-d43e-421a-b6bf-0727141954ed)

As you can see, one of these nights it has performed 16 defrost cycles (marked in red) which have triggered at least two DHW recharges (marked in yellow, let's not consider the first one, the tank temperature was already low) at a time that was not exceptional for the operation, with an external temperature between a maximum of 3 degrees and a minimum of -1. In short, the temperature was certainly rigid, but not extreme either, and all these defrosts seem a bit too many to me, spaced an hour apart, lasting 5 minutes and each subtracting 2 degrees from the technical accumulation. I also noticed that during the defrost phases the BPV valve remains open and the defrost flow (with a higher temperature, still in the operating range of the radiant) passes through the floor, when perhaps for a few minutes a direct recirculation would have made more sense, with less subtraction from the technical accumulation. The machine has the original firmware, do you think it is worth using a Daikin center to have the update to the latest useful version, if something on this front has changed? I remember that it is a Daikin Rotext HPSU Compact 508 H/C BIV 8 Kw from 2015, still with its original firmware.

The last question is: having all the data managed via home automation, has the issue of creating an algorithm already been addressed that, taking into account a whole series of data (internal and external temperature, water flow and return, desired temperature, etc.) sets the flow temperature in a variable and efficient way, perhaps taking into account some presets aimed at optimizing efficiency? Because I was thinking about it, but I would avoid reinventing the wheel if someone has already done it and it works better than mine ;)

Thanks,
D.

**Reply by wrfz** [reply]
First of all: Please do not mix topics in the future. If you have questions that have nothing to do with the current topic, please start a new discussion. A huge discussion with different topics is not very helpful.

**Reply by wrfz** [reply]
HPSU Compact (referred to as V4 by Daikin) has at least two known issues related to defrosting:

The HPSU uses hot water from the DHW tank to defrost the evaporator. Ideally, both valves (DHW mixer and bypass) would open, allowing the water from the hot water tank to transfer energy to the refrigerant circuit via the plate heat exchanger. I’m not sure if this was an intentional feature by the developers or a mistake. In any case, the bypass valve closes at that moment, which causes part of the hot water to be pushed into the outdoor unit while simultaneously sending additional "expensive" hot water into the underfloor heating system. In my opinion, this is a flaw.

To address this, I implemented a feature in our CAN component called "Defrost Optimization." When enabled, the operating mode temporarily switches to "Summer" during the defrost phase and reverts to "Heating" afterward. This ensures that the bypass valve remains open during the defrost phase, preventing hot water from being directed into the underfloor heating system. This adjustment should reduce the thermal loss per defrost cycle from approximately 2K to about 0.7K.

(Sorry for a screenshot in German language)
![grafik](https://github.com/user-attachments/assets/453b1928-806e-43a4-aa12-cabc848f5a6d)

**Reply by wrfz** [reply]
The second issue with the HPSU Compact control system is that it defrosts too frequently. According to my analysis, in certain situations, it defrosts as often as every 25 minutes, even though the evaporator is only partially iced, with at least 50% of the evaporator surface still ice-free. Statistics show that at the time of the defrost cycle, the dynamically calculated COP does not experience any reduction. Therefore, the defrost process is simply unnecessary in many cases.

There are likely several reasons for this:

- Rotex managed to design a "functional" device with minimal use of sensors (which is quite an achievement in itself).
- The refrigerant routing in the evaporator is probably not optimal (at least some aspects seem unnecessarily complicated to me).
- The detection of a frozen evaporator likely relies on the delta between the outdoor temperature sensor and the evaporator sensor. Since the evaporator doesn’t freeze uniformly (and the evaporator sensor is probably located in an ice-free spot), the defrost cycle is triggered prematurely.

In the Telegram group on the [Daikin-Rotex-HPSU-CAN-Serial - Startpage](https://github.com/Trunks1982/Daikin-Rotex-HPSU-CAN-Seriell/) (see at the very bottom), various ideas are being discussed and partially tested:

1. Relocating one of the two sensors to reduce the delta, so that defrosting is triggered less frequently. This idea is already a few years old and has been implemented by various people who successfully reduced the defrost cycles. For others, however, it showed little success.
2. Manipulating the resistance of the evaporator or outdoor temperature sensor via an ESP to intentionally delay the defrost cycle (e.g., only triggering it when the COP value starts to drop significantly). This is a completely new idea and still needs to prove whether it will work effectively.

**Reply by wrfz** [reply]
Regarding the firmware version: I have repeatedly inquired with Daikin technical support and always received the same response (which I find very hard to believe): There is only one version!

**Reply by wrfz** [reply]
Regarding control via an adaptive flow temperature, you’ll also find an HA template sensor in the Telegram group, which you can use to control the ESP through HA automation.

**_Caution_**: You should ensure that adjustments are made no more than once per hour. Changing any settings in the HPSU (even via ESP) likely results in the settings being persisted. Since the data is most likely stored on an EEPROM, and most EEPROMs typically endure a maximum of about 100,000 write cycles, it’s advisable to avoid making frequent changes to the HPSU settings!

**Reply by DanieleNeo** [reply]
> HPSU Compact (referred to as V4 by Daikin) has at least two known issues related to defrosting:
> 
[...]
> 
> (Sorry for a screenshot in German language)

no problem for the screenshot in German language, having spent the last few evenings working on the multilanguage implementation I have now learned the fundamentals :)

I tried the optimized defrost feature, and I can confirm that now defrost cycle consume less heat from the thermal storage, triggering less DHW recharges. A really great idea this to "trick" the heat pump by changing the operating mode during the defrost

D.

**Reply by DanieleNeo** [reply]
> The second issue with the HPSU Compact control system is that it defrosts too frequently. According to my analysis, in certain 
[...]

Here in Italy there are many users who are afflicted by this problem, some have even made videos showing a defrost phase without any ice (https://www.youtube.com/watch?v=Ui2ZJ1OBdC4) and Daikin Italia was also interested, and after a couple of inspections they ruled that the device is working correctly. I find all this absurd, given that one of the purposes of the heat pump is to heat efficiently, that there are energy-intensive phases uselessly invoked, just as I find it absurd that Daikin does not admit the problem and does not release a software update that can at least mitigate the problem, such as allowing the user to choose from which source (even from the floor for example, taking advantage of its inertia) to subtract the energy to perform the defrost phase, and for how long. It's probably as you say, they wanted to make a "simple" device and therefore they saved on the sensors, in fact it seems that users who have the 516 do not suffer from these problems (and if I remember correctly the external unit of that device had a few more sensors)

> 1. Relocating one of the two sensors to reduce the delta, so that defrosting is triggered less frequently. This idea is already a few years old and has been implemented by various people who successfully reduced the defrost cycles. For others, however, it showed little success.

Yes, this solution has also been discussed on a forum here in Italy, someone has done it and actually solved it. Personally I would not like to resort to such a degree of hacking, also because I have a BIV version and I need the heat to heat the floor, when I use the thermo-fireplace, to be subtracted from the thermal accumulation, otherwise I wouldn't know what to do with it just for DHW. Maybe I'm talking about the implementation I read about on the Italian forum, I don't remember the details and which probes they had moved, but they actually claimed that this was not possible if you had a second integrated heat generator

> 2. Manipulating the resistance of the evaporator or outdoor temperature sensor via an ESP to intentionally delay the defrost cycle (e.g., only triggering it when the COP value starts to drop significantly). This is a completely new idea and still needs to prove whether it will work effectively.

This is a dynamic solution that I definitely prefer, implementing additional sensors (I don't know what type at the moment) to understand if a defrost is actually necessary, acting on the external temperature and evaporator probes. Maybe it could also be enough to create a model based on temperature and humidity, in the end it should be "simple" to hypothesize how much ice has formed considering the humidity, temperature and execution time from the last useful state (be it a favorable temperature or a defrost cycle). I don't know how reliable it is to rely on the COP, in the end it is relative information, which undergoes fluctuations, and you could risk realizing that the machine does not work when the battery is completely frozen. It would also be useful to know how to act on the real sensors, to activate the defrost and at the same time not generate malfunctions (if I'm not mistaken with these temperatures in addition to starting the defrost the machine calculates the climate curve, and perhaps even more). Anyway thanks for the information, I'll take a look at the telegram group.

D.

**Reply by DanieleNeo** [reply]
> Regarding the firmware version: I have repeatedly inquired with Daikin technical support and always received the same response (which I find very hard to believe): There is only one version!

Either they are the only ones who do perfect things (and I doubt it), or it makes no sense. I have the 366 rev02g for the controller and 1.08.98 for RTX, so it's the latest?

**Reply by DanieleNeo** [reply]
> Regarding control via an adaptive flow temperature, you’ll also find an HA template sensor in the Telegram group, which you can use to control the ESP through HA automation.
> 
> **_Caution_**: You should ensure that adjustments are made no more than once per hour. Changing any settings in the HPSU (even via ESP) likely results in the settings being persisted. Since the data is most likely stored on an EEPROM, and most EEPROMs typically endure a maximum of about 100,000 write cycles, it’s advisable to avoid making frequent changes to the HPSU settings!

Too bad, it would have been nice, persisting the information only when the machine turns off or restarts was asking too much, or providing a save function. It's a real shame, because by applying efficient algorithms the software obsolence could have been overcome, at least for normal use. I was also thinking, since I have two towel warmers in the bathrooms, of providing a program that would exclude the floor rings and send water at 50 degrees to the towel warmers, at least when you have to put something out to dry in an emergency (like motorcycle gloves after a rain) you can...

**Reply by wrfz** [reply]
>  It would also be useful to know how to act on the real sensors, to activate the defrost and at the same time not generate malfunctions (if I'm not mistaken with these temperatures in addition to starting the defrost the machine calculates the climate curve, and perhaps even more).

We may never find out with 100% certainty. However, by analyzing the data provided by the device, we might be able to achieve good results.

To my understanding, once an external air temperature sensor is installed and configured, the air temperature sensor in the outdoor unit is used only for the defrosting process. The heating curve (and thus the flow temperature in the weather-compensated heating circuit function) is calculated based on the external sensor.

**Reply by wrfz** [reply]
> > Regarding the firmware version: I have repeatedly inquired with Daikin technical support and always received the same response (which I find very hard to believe): There is only one version!
> 
> Either they are the only ones who do perfect things (and I doubt it), or it makes no sense. I have the 366 rev02g for the controller and 1.08.98 for RTX, so it's the latest?

I and at least one other user in the group have exactly the same n versions

**Reply by wrfz** [reply]
> Too bad, it would have been nice, persisting the information only when the machine turns off or restarts was asking too much, or providing a save function.

Unfortunately, the V4 does not have the capability to persist data or trigger a restart. The only option is to turn off the power, and in that case, nothing will be persisted.

The assumption that the EEPROM could be damaged is just my guess; I might be wrong here. However, until Daikin disproves this point, I wouldn’t take any chances and would recommend limiting the flow temperature adjustments to a maximum of once per hour.

That being said, I don’t see a significant issue here. Since underfloor heating is very slow to respond and takes hours to adjust room temperature to match the flow temperature, it doesn’t make much sense to change the flow temperature at minute intervals.

There is one important point to consider, though:
If the flow temperature is reduced too drastically in one step (causing a difference of 3K or more between the current flow temperature and the target flow temperature), the HPSU shuts off the compressor. The ESP/HA algorithm should always ensure that the temperature is reduced in several smaller steps.

For this purpose, I plan to add some functionality to the CAN component so it can handle this process autonomously.

**Comment by DanieleNeo** [comment]
I tried last version and I can confirm that the error never occurs.

@wrfz @Trunks1982 as I said before I have currently forked the repository and I'm working on my own copy localized in Italian, do you think it could be useful to centralize the textual resources in a dictionary (some are hard coded in the code) and realize a multilingual version German/English/Italian? Because I was thinking about it, if you think it's useful I can spend some time on it, let me know, happy to colaborate.

Thanks,
D.

**Reply by wrfz** [reply]
I’ll take a look to see if I can implement localization of a dictionary with reasonable effort.

**Reply by DanieleNeo** [reply]
I'll answer the last one and then I'll answer the others (you're right in saying that maybe it's better to diversify the topics... but should I answer here now?), I could work on the localization, if you think it's useful.

**Reply by DanieleNeo** [reply]
> I’ll take a look to see if I can implement localization of a dictionary with reasonable effort.

Hi @wrfz,

I have done the work, I opened the pull request https://github.com/Trunks1982/Daikin-Rotex-HPSU-CAN-Seriell/pull/65 waiting for review and approval. I'll start with same implementation on the serial component, even if in a lighter version (the dictionary for the cpp part is not needed in that case).

D.
